
Reply to Somayah:

> Leveraging deep learning to study splicing mechanism is not a novel topic.
> The authors reviewed the related works in the Introduction. However, some recent works are not included. Add the following citations:
> a. Predicting splicing from primary sequence with deep learning. Cell
> b. Discerning novel splice junctions derived from RNA-seq alignment: a deep learning approach. BMC genomics

### TODO
- convert processed data to csv for Ashraf
- access Ashraf's workstation via ssh (see mail)
- try to access IBEX for bigger computational things
- build up some presentation for the done work

- Vary position of flatten layer

### Readworthy links
- [spliceRover](http://bioit2.irc.ugent.be/rover/splicerover/)

### Readworthy papers:

- DiProDB 
- Do PCA with 15 best features
- electron ion interaction potential
  - [Other](https://www.researchgate.net/publication/6239442_A_coding_measure_scheme_empoying_electron-ion_interaction_pseudo_potential_EIIP)
  - [Nucleotide values](https://www.sciencedirect.com/science/article/pii/030326479090013Q)

- Checkout AAindex: Amino acid index database
- checkout APDbase
- See reddit for more
